tests_common/pytest_plugin.py

```diff
@@ -31,6 +31,18 @@
 import pytest
 import time_machine
 
+# NOTE: DO NOT IMPORT AIRFLOW THINGS HERE!
+#
+# This plugin is responsible for configuring Airflow correctly to run tests.
+# Importing Airflow here loads Airflow too eagerly and break the configurations.
+# Instead, import what you want lazily inside a fixture function.
+#
+# Be aware that many things in tests_common also indirectly imports Airflow, so
+# those modules also should not be imported globally.
+#
+# (Things in the TYPE_CHECKING block are fine because they are not actually
+# imported at runtime; those imports are only hints to the type checker.)
```

Comment:
Is it worth addming a check? Something like this?
```suggestion
# imported at runtime; those imports are only hints to the type checker.)

assert "airflow" not in sys.modules
```

Reply:
Absolutely - we already have this in conftest.py:

```
# We should set these before loading _any_ of the rest of airflow so that the
# unit test mode config is set as early as possible.
assert "airflow" not in sys.modules, "No airflow module can be imported before these lines"
```

Reply:
Probably. If our test setup gets more complicated people might need some additional pointers toward these comments. This assert showing up in the error log can help. I’ll add it with a more helpful error message.